torch/utils/_benchmark/utils/common.py

```diff
@@ -50,6 +52,7 @@ def __init__(
         self.num_threads = num_threads
         self.stmt = stmt
         self.metadata = metadata
+        self._has_cache_warning = False
```

Comment:
I'm reluctant to formalize warnings too much right now, as it adds a lot of extra complexity. (For instance as is `_has_cache_warning` doesn't serialize/deserialize properly.) Let's just stick to adding an entry in `self._warnings`.

Reply:
okay!